src/index.ts

```diff
@@ -165,6 +200,99 @@ const createCondaMetaFile = (
   }
 };
 
+export const bootstrapFromEmpackPackedEnvironment = async (
+  packagesJsonUrl: string,
+  verbose: boolean = true,
+  skipLoadingSharedLibs: boolean = false,
+  Module: any
+) => {
+  if (verbose) {
+    console.log('fetching packages.json from', packagesJsonUrl);
+  }
+
+  // fetch json with list of all packages
+  let empackEnvMeta = await fetchJson(packagesJsonUrl);
+  let allPackages: IEmpackEnvMetaPkg[] = empackEnvMeta.packages;
+  let prefix = empackEnvMeta.prefix;
+
+  console.log('allPackages', allPackages);
+
+  if (verbose) {
+    console.log('makeDirs');
+  }
+  //Module.create_directories("/package_tarballs");
+
+  // enusre there is python and split it from the rest
+  if (verbose) {
+    console.log('splitPackages');
+  }
+  let splitted: ISplittedPackages = splitPackages(allPackages);
+  let packages = splitted.packages;
+  let pythonPackage: IEmpackEnvMetaPkg = splitted.pythonPackage;
+  let pythonVersion = pythonPackage.version.split('.').map(x => parseInt(x));
+
+  // fetch init python itself
+  console.log('--bootstrap_python');
+  if (verbose) {
+    console.log('bootstrap_python');
+  }
+  //TO Do
+  //let python_is_ready_promise = bootstrap_python(prefix, package_tarballs_root_url, pythonPackage, verbose);
+
+  // create array with size
+  if (verbose) {
+    console.log('fetchAndUntarAll');
+  }
+
+  let sharedLibs = await Promise.all(
+    packages.map(pkg => installCondaPackage(prefix, pkg.url, Module.FS, verbose))
+  );
+
+  console.log(`sharedLibs`, sharedLibs);
+
+ /* if (!skipLoadingSharedLibs) {
+    loadShareLibs(packages, sharedLibs, prefix, pythonVersion);
+  }*/
+};
+
+const splitPackages = (packages: IEmpackEnvMetaPkg[]) => {
+  let pythonPackage: IEmpackEnvMetaPkg | undefined = undefined;
+  for (let i = 0; i < packages.length; i++) {
+    if (packages[i].name == 'python') {
+      pythonPackage = packages[i];
+      packages.splice(i, 1);
+      break;
+    }
+  }
+  if (pythonPackage == undefined) {
+    throw new Error('no python package found in package.json');
```

Comment:
We may have cases with environments where Python is not installed, and we want to support those cases. As I understand your current implementation does not support that?

Also I'm confused by this error message, maybe you mean `empack lock file` not `package.json`?